runtime/vm/ClassInitialization.cpp

```diff
@@ -386,6 +386,19 @@ classInitStateMachine(J9VMThread *currentThread, J9Class *clazz, J9ClassInitStat
 				if (NULL == initializationLock) {
 					goto done;
 				}
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+				/* Preparation is the earliest point where the defaultValue would needed. 
+				 * I.e pre-filling static fields. Therefore, the defaultValue must be allocated at 
+				 * the end of verification 
+				 */
+				j9object_t defaultValue = vm->memoryManagerFunctions->J9AllocateObject(currentThread, clazz, J9_GC_ALLOCATE_OBJECT_TENURED | J9_GC_ALLOCATE_OBJECT_NON_INSTRUMENTABLE);
+				if (NULL == defaultValue) {
+					setHeapOutOfMemoryError(currentThread);
+					goto done;
+				}
+				clazz->flattenedClassCache[0].clazz = (J9Class *)defaultValue;
+				clazz = VM_VMHelpers::currentClass(clazz);
```

Comment:
`initializationLock` needs to be push in a special frame before the allocation and pop'ed after the allocation.

See above for an example

Reply:
```
initializationLock = POP_OBJECT_IN_SPECIAL_FRAME(currentThread);
clazz = VM_VMHelpers::currentClass(clazz);
```
should be after the allocation

Reply:
Got it, like this?